packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/MethodCallHandlerImpl.java

```diff
@@ -47,10 +53,36 @@ public void onMethodCall(MethodCall call, MethodChannel.Result result) {
         (ShortcutManager) context.getSystemService(Context.SHORTCUT_SERVICE);
     switch (call.method) {
       case "setShortcutItems":
-        List<Map<String, String>> serializedShortcuts = call.arguments();
-        List<ShortcutInfo> shortcuts = deserializeShortcuts(serializedShortcuts);
-        shortcutManager.setDynamicShortcuts(shortcuts);
-        break;
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.N_MR1) {
+          List<Map<String, String>> serializedShortcuts = call.arguments();
+          List<ShortcutInfo> shortcuts = deserializeShortcuts(serializedShortcuts);
+
+          Executor uiThreadExecutor = new UiThreadExecutor();
+          ThreadPoolExecutor executor =
+              new ThreadPoolExecutor(
+                  0, 1, 1, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>());
+
+          executor.execute(
+              () -> {
+                boolean dynamicShortcutsSet = false;
+                try {
+                  shortcutManager.setDynamicShortcuts(shortcuts);
+                  dynamicShortcutsSet = true;
+                } catch (Exception e) {
+                  result.error(
+                      "quick_action_setshortcutitems_failure",
+                      "Exception thrown when setting dynamic shortcuts",
+                      null);
```

Comment:
This is on the wrong thread; any interaction with `result` needs to be (on current stable) on the main thread.